api/client/build.go

```diff
@@ -265,6 +268,18 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	headers.Add("X-Registry-Config", base64.URLEncoding.EncodeToString(buf))
 	headers.Set("Content-Type", "application/tar")
 
+	// collect all the build-time environment variables for the container
+	buildArgs := runconfig.ConvertKVStringsToMap(flArg.GetAll())
+	buf, err = json.Marshal(buildArgs)
+	if err != nil {
+		return err
+	}
+	headers.Add("X-Docker-BuildArgs", base64.URLEncoding.EncodeToString(buf))
+
+	if context != nil {
```

Comment:
I know its premature since we're still in design review, but is this "context" stuff part of this PR or due to some other issue?